test/OrchardCore.Tests/Apis/ContentManagement/DeploymentPlans/BlogPostCreateDeploymentPlanTests.cs

```diff
@@ -0,0 +1,152 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Tests.Apis.Context;
+using Xunit;
+using YesSql;
+
+namespace OrchardCore.Tests.Apis.ContentManagement.DeploymentPlans
+{
+    public class BlogPostCreateDeploymentPlanTests
+    {
+        [Fact]
+        public async Task ShouldCreateNewPublishedContentItemVersion()
+        {
+            using (var context = new BlogPostDeploymentContext())
+            {
+                // Setup
+                await context.InitializeAsync();
+
+                // Act
+                var recipe = context.GetContentStepRecipe(context.OriginalBlogPost, jItem =>
+                {
+                    jItem[nameof(ContentItem.ContentItemVersionId)] = "newVersion";
+                    jItem[nameof(ContentItem.DisplayText)] = "new version";
+                });
+
+                await context.PostRecipeAsync(recipe);
+
+                // Test
+                using (var shellScope = await BlogPostDeploymentContext.ShellHost.GetScopeAsync(context.TenantName))
+                {
+                    await shellScope.UsingAsync(async scope =>
+                    {
+                        var session = scope.ServiceProvider.GetRequiredService<ISession>();
+                        var blogPosts = await session.Query<ContentItem, ContentItemIndex>(x =>
+                            x.ContentType == "BlogPost").ListAsync();
+
+                        Assert.Equal(2, blogPosts.Count());
+
+                        var originalVersion = blogPosts.FirstOrDefault(x => x.ContentItemVersionId == context.OriginalBlogPostVersionId);
+                        Assert.False(originalVersion?.Latest);
+                        Assert.False(originalVersion?.Published);
+
+                        var newVersion = blogPosts.FirstOrDefault(x => x.ContentItemVersionId == "newVersion");
+                        Assert.Equal("new version", newVersion?.DisplayText);
+                        Assert.True(newVersion?.Latest);
+                        Assert.True(newVersion?.Published);
+                    });
```

Comment:
Would appreciate some input from anyone on how I am building these integration tests up.

I've co-opted @Jetski5822 GraphQL API tests, and introduced access to the DI container of the built tenant.
This is used to configure remote recipe deployment (as we have no api for doing that - it could be done via a custom recipe, but...

For the general Act of the test I am using the API's or controller endpoints.

For testing it however, I'm resolving the `YesSql.ISession` and querying the database directly, as we have no API endpoint that would allow me to test it otherwise.

I think it's proving useful, as it's identified a number of issues, particularly in the `ContentApiController` where records were being incorrectly stored, and sometimes duplicated (fixed)

But opinions of others here would be helpful if they think it is an acceptable integration testing pattern

Reply:
Funny, we communicate through the API with a tenant on which we can create a scope ;)

I'm okay with this pattern where we build a shell scope but only to check the results.